### PR TITLE
[Secrets Sync] enable access to Sync clients page for HVD clusters

### PR DESCRIPTION
**Pull request by noelledaley**
### :hammer_and_wrench: Description
- show sync clients in client count nav bar if there's a sync client history OR if the feature is **able to be** activated
- show sync charts on overview page if sync client history OR if the feature **is** activated 

🧹 move client count nav bar to its own component

### :camera_flash: Screenshots
image

### :building_construction: How to Build and Test the Change
For OSS
- sync client charts and tabs should be hidden
For HVD (most easily accomplished by pulling this down locally and hard-coding `isHvdManaged = true` in the `flags` service)
- sync client tab is shown
- charts are hidden unless there's a sync client history

### CODE REVIEW

File: ui/app/components/clients/page/counts.hbs [hunk 1]
```diff
@@ -151,32 +151,7 @@
         </Hds::Alert>
       {{/if}}
 
-      <nav class="tabs has-bottom-margin-s" aria-label="navigation for managing client counts">
```
Comment (noelledaley) [inline ui/app/components/clients/page/counts.hbs:154]: this file & its associated test were getting a bit hairy, so moved it into a separate component 🧹

File: ui/app/templates/vault/cluster/clients/counts.hbs [hunk 2]
```diff
@@ -8,7 +8,6 @@
   @activityError={{this.model.activityError}}
   @config={{this.model.config}}
   @endTimestamp={{this.model.endTimestamp}}
-  @isSecretsSyncActivated={{this.model.isSecretsSyncActivated}}
```
Comment (noelledaley) [inline ui/app/templates/vault/cluster/clients/counts.hbs:11]: this was never used 🧹

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 3]
```diff
@@ -40,12 +40,6 @@ module('Acceptance | clients | overview', function (hooks) {
     timestamp.now.restore();
   });
 
-  test('it should render the correct tabs', async function (assert) {
```
Comment (noelledaley) [inline ui/tests/acceptance/clients/counts/overview-test.js:43]: this is now covered by the `nav-bar-tests` integration tests 😀

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 4]
```diff
@@ -59,28 +52,19 @@ export default class ClientsCountsRoute extends Route {
...
+    if (this.version.isCommunity) return false;
 
-    // if feature is not in license, it's definitely not activated
-    if (!this.version.hasSecretsSync) return false;
+    const isManaged = this.flags.isManaged;
+    const onLicense = this.version.hasSecretsSync;
 
-    // otherwise check explicitly if the feature has been activated
-    const activatedFeatures = await this.getActivatedFeatures();
-    return activatedFeatures.includes('secrets-sync');
```
Comment (noelledaley) [inline ui/app/routes/vault/cluster/clients/counts.ts:83]: removing this line is what gives HVD clusters access to these pages and charts. now, the new way we allow access is:

```typescript
    // if there is a history of sync clients, show it
    if (activity && activity?.total?.secret_syncs > 0) return true;

    // otherwise, show the tab based on the cluster type and license
    if (this.version.isCommunity) return false;
    if (this.version.isCommunity) return false;

    const isManaged = this.flags.isManaged;
    const onLicense = this.version.hasSecretsSync;

    // we can't tell if HVD clusters have the feature or not, so we show it by default
    // if the cluster is not HVD, show the tab if the feature is on the license
    return isManaged || onLicense;
```

i renamed this to `showSecretsSync` because it doesn't actually check whether the feature is activated. 

TODO:
- show sync charts only if feature is activated

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 5]
```diff
@@ -59,28 +52,19 @@ export default class ClientsCountsRoute extends Route {
...
-        .adapterFor('application')
-        .ajax('/v1/sys/activation-flags', 'GET', { unauthenticated: true, namespace: null });
-      return resp.data?.activated;
-    } catch (error) {
-      if (DEBUG) console.error(error); // eslint-disable-line no-console
-      return [];
-    }
-  }
+  showSecretsSync(activity: ClientsActivityModel | undefined): boolean {
+    // if there is a history of sync clients, show it
```
Comment (hellobontempo) [inline ui/app/routes/vault/cluster/clients/counts.ts:56]: technically this isn't just historical data since the end time we initially query is the current month. definitely semantics, but just wanted to point that out! I might suggest rewording to something like `if there is any sync client data, show it`

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 6]
```diff
@@ -59,28 +52,19 @@ export default class ClientsCountsRoute extends Route {
...
-    // if feature is not in license, it's definitely not activated
-    if (!this.version.hasSecretsSync) return false;
+    const isManaged = this.flags.isManaged;
+    const onLicense = this.version.hasSecretsSync;
 
-    // otherwise check explicitly if the feature has been activated
-    const activatedFeatures = await this.getActivatedFeatures();
-    return activatedFeatures.includes('secrets-sync');
+    // we can't tell if HVD clusters have the feature or not, so we show it by default
+    // if the cluster is not HVD, show the tab if the feature is on the license
```
Comment (hellobontempo) [inline ui/app/routes/vault/cluster/clients/counts.ts:66]: great comments! 👏

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 7]
```diff
@@ -59,28 +52,19 @@ export default class ClientsCountsRoute extends Route {
...
 
-  async isSecretsSyncActivated(activity: ClientsActivityModel | undefined) {
-    // if there are secrets, the feature is activated
-    if (activity && activity.total?.secret_syncs > 0) return true;
+    // otherwise, show the tab based on the cluster type and license
+    if (this.version.isCommunity) return false;
 
-    // if feature is not in license, it's definitely not activated
-    if (!this.version.hasSecretsSync) return false;
+    const isManaged = this.flags.isManaged;
```
Comment (hellobontempo) [inline ui/app/routes/vault/cluster/clients/counts.ts:62]:
```suggestion
    const isHvdManaged = this.flags.isHvdManaged;
```

File: ui/app/routes/vault/cluster/clients/counts.ts (hunk 4 above)
Comment (hellobontempo) [reply ui/app/routes/vault/cluster/clients/counts.ts:83]: I think above some of the conditionals are copy pasta'd but this made sense reading the method! 
And good call on renaming the method!

File: ui/app/components/clients/page/counts.hbs [hunk 8]
```diff
@@ -151,32 +151,7 @@
...
-            </li>
-          {{/if}}
-          <li>
-            <LinkTo @route="vault.cluster.clients.counts.acme" data-test-tab="acme">
-              ACME clients
-            </LinkTo>
-          </li>
-        </ul>
-      </nav>
+      <Clients::Counts::NavBar @activity={{@activity}} />
```
Comment (hellobontempo) [inline ui/app/components/clients/page/counts.hbs:154]: Good idea :bulq to make this a component inside a `counts/` directory. I think it makes sense since it's probably only ever going to be rendered inside this component. 

Could we also move `<Clients::Error>` to that directory as well? (`<Clients::NoData>` is used in the dashboard.hbs file, but the error template is just for this component)

File: ui/app/components/clients/page/sync.hbs [hunk 9]
```diff
@@ -2,7 +2,7 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: BUSL-1.1
 ~}}
-{{#if @isSecretsSyncActivated}}
+{{#if @showSecretsSync}}
```
Comment (hellobontempo) [inline ui/app/components/clients/page/sync.hbs:5]: I'm not sure we want this conditional here 🤔 `@showSecretsSync` already wraps the tabs so we're hiding this route in cases where secret syncing is not possible. So I think this should be updated to just check for activation so it can render the `No data is available because Secrets Sync has not been activated.` empty state below. 

No more granular checks are needed because the template renders a separate empty state for 0 sync clients.

File: ui/app/components/clients/page/counts.hbs (hunk 8 above)
Comment (noelledaley) [reply ui/app/components/clients/page/counts.hbs:154]: gladly!

File: ui/app/components/clients/page/sync.hbs (hunk 9 above)
Comment (noelledaley) [reply ui/app/components/clients/page/sync.hbs:5]: ah yes, this is partially what i meant by "add more granular checks" in my TODO comment above, but i can see that was not obvious at all. 😂

appreciate you noticing / clarifying!

File: ui/app/routes/vault/cluster/clients/counts.ts (hunk 4 above)
Comment (noelledaley) [reply ui/app/routes/vault/cluster/clients/counts.ts:83]: ah, WRT more granular checks i was thinking about this situation:

- no history of sync clients
- not activated

i believe we want to hide the sync client charts in that case because there's nothing to show. this is what's outlined in the tests today, too:
https://github.com/hashicorp/vault/blob/af3e6dd2a71b0b881f81bfcc0be8acdb69ccf292/ui/tests/acceptance/clients/counts/overview-test.js#L281-L291

but yes, your comment on the sync page component is also correct!

Comment (hellobontempo) [reply ui/app/routes/vault/cluster/clients/counts.ts:83]: Okay gotcha. We just want to make sure we don't "hide" the charts based on 0 sync clients (no history of sync clients) so I just wanted to confirm that the empty chart/empty state logic would be staying the same 😄

File: ui/app/components/clients/page/overview.ts [hunk 10]
```diff
@@ -4,5 +4,13 @@
...
 import ActivityComponent from '../activity';
+import { service } from '@ember/service';
+import type FlagsService from 'vault/services/flags';
 
-export default class ClientsOverviewPageComponent extends ActivityComponent {}
+export default class ClientsOverviewPageComponent extends ActivityComponent {
+  @service declare readonly flags: FlagsService;
+
+  get hasSyncHistoryOrIsActivated() {
+    return this.args.activity.total.secret_syncs > 0 || this.flags.secretsSyncIsActivated;
```
Comment (hellobontempo) [inline ui/app/components/clients/page/overview.ts:14]: 🤔 Hmm, maybe I wasn't clear in my previous comments. But 0 clients in this context is still an important metric so we should remove the first part of this conditional.

Comment (noelledaley) [reply ui/app/components/clients/page/overview.ts:14]: oh oh oh. yes, i understand what you mean. should we check if `secret_syncs !== undefined` then? or `!== null`? let's chat in slack about this.

File: ui/app/components/clients/page/overview.hbs [hunk 11]
```diff
@@ -4,7 +4,7 @@
 ~}}
 
 <Clients::RunningTotal
-  @isSecretsSyncActivated={{@isSecretsSyncActivated}}
+  @showSecretsSync={{this.flags.secretsSyncIsActivated}}
```
Comment (hellobontempo) [inline ui/app/components/clients/page/overview.hbs:7]: Sorry, I know this suggestion may be a pain 🙈  

Since this arg comes from `this.flags.secretsSyncIsActivated` I think keeping it`@isSecretsSyncActivated` is important here (and below) to distinguish it from the boolean that comes from the route's `showSecretsSync` method. This way a global search of `showSecretsSync` will only show the relevant args, which sets us up for future refactors/renaming (rather than having to fully rehash this logic, which we've invested a lot of energy to sort out 😅). An additional benefit, is the template will be more explicit because it will explain that the wrapped elements are reliant on activation.
```suggestion
  @isSecretsSyncActivated={{this.flags.secretsSyncIsActivated}}
```

File: ui/app/templates/vault/cluster/clients/counts/overview.hbs [hunk 12]
```diff
@@ -5,7 +5,7 @@
 
 <Clients::Page::Overview
   @activity={{this.model.activity}}
-  @isSecretsSyncActivated={{this.model.isSecretsSyncActivated}}
+  @showSecretsSync={{this.model.showSecretsSync}}
```
Comment (hellobontempo) [inline ui/app/templates/vault/cluster/clients/counts/overview.hbs:8]: I _think_ this arg can be deleted because the child components get their boolean from the flag service 🧹

File: ui/app/templates/vault/cluster/clients/counts/sync.hbs [hunk 13]
```diff
@@ -4,7 +4,7 @@
 ~}}
 
 <Clients::Page::Sync
-  @isSecretsSyncActivated={{this.model.isSecretsSyncActivated}}
+  @showSecretsSync={{this.model.showSecretsSync}}
```
Comment (hellobontempo) [inline ui/app/templates/vault/cluster/clients/counts/sync.hbs:7]: Same comment here as above, I think this can be deleted

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 14]
```diff
@@ -153,79 +148,98 @@ module('Acceptance | clients | overview', function (hooks) {
...
+
+    test('filters by auth method correctly', async function (assert) {
+      const response = await this.store.peekRecord('clients/activity', 'some-activity-id');
+      const topNamespace = response.byNamespace[0];
+      const topMount = topNamespace.mounts[0];
+
+      await clickTrigger();
+      await searchSelect.options.objectAt(0).click();
+      await settled();
+      debugger;
```
Comment (hellobontempo) [inline ui/tests/acceptance/clients/counts/overview-test.js:198]: These tests are likely still WIP but I noticed the debugger here, just FYI!

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 15]
```diff
@@ -153,79 +148,98 @@ module('Acceptance | clients | overview', function (hooks) {
...
-      .includesText(`${formatNumber([topMount.clients])}`, 'top attribution clients accurate');
-
-    let expectedStats = {
-      Entity: formatNumber([topNamespace.entity_clients]),
-      'Non-entity': formatNumber([topNamespace.non_entity_clients]),
-      ACME: formatNumber([topNamespace.acme_clients]),
-      'Secret sync': formatNumber([topNamespace.secret_syncs]),
-    };
-    for (const label in expectedStats) {
+  module('filtering charts', function () {
```
Comment (hellobontempo) [inline ui/tests/acceptance/clients/counts/overview-test.js:158]: Was the decision to convert to modules to try and address the flakiness? In my experience, these tests weren't flaky before 🤔 , so if that isn't the case then could this refactor happen in a separate PR? 

I completely agree with refactoring these into modules! But a separate PR would be make this one easier to review and check that test coverage has been added for the relevant logic updates. Historically, we've also had folks accidentally remove important test assertions for this area of the code base due to the intricacies of this dang dashboard 😞

File: ui/app/components/clients/page/overview.hbs (hunk 11 above)
Comment (noelledaley) [reply ui/app/components/clients/page/overview.hbs:7]: not a pain Claire, i absolutely welcome this kind of feedback! you're 100% right -- keeping the OG name _is_ more clear and distinguishing the arg from the arg passed down from the model is important. will update this pronto.

File: ui/tests/acceptance/clients/counts/overview-test.js (hunk 15 above)
Comment (noelledaley) [reply ui/tests/acceptance/clients/counts/overview-test.js:158]: that's correct, i moved them into separate tests to try to isolate the failures. what happened here was that i switched to using the `sync` mirage handler instead of the `clients` one at the top-level `beforeEach` hooks because making assertions on the secrets sync charts required secrets sync being included in the `/activated-features` response, which the `clients` handler didn't have. similarly, i assumed the `sync` handler would also be better to use here since it would include sync client data to populate the charts. however when i made this switch, the tests failed inconsistently, so i split the tests up to try and figure out if it was only specific aspects of the tests or if it was the whole ding-dang thing. 😂

as it turns out, the root of the problem lies somewhere in the differences between the `sync` and `clients` handlers, because yeah, splitting up the module didn't help much at all. i am reverting this change today and am investigating other options (potentially leaving the sync assertions out of these tests and instead making those assertions in a completely separate module so we can use the `sync` handler instead of `clients`). 🤔

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 16]
```diff
@@ -238,15 +233,7 @@ module('Acceptance | clients | overview | sync in license, activated', function
...
-    clientsHandler(this.server);
-    this.store = this.owner.lookup('service:store');
-
-    // add feature to license
-    this.server.get('/sys/license/features', () => ({ features: ['Secrets Sync'] }));
-    // activate feature
-    this.server.get('/sys/activation-flags', () => ({
-      data: { activated: ['secrets-sync'], unactivated: [] },
-    }));
+    syncHandler(this.server);
```
Comment (noelledaley) [inline ui/tests/acceptance/clients/counts/overview-test.js:236]: the sync handler sets up all the mocked endpoints that we had before anyway 😀

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 17]
```diff
@@ -288,12 +278,11 @@ module('Acceptance | clients | overview | sync in license, not activated', funct
...
-  test('it should hide secrets sync charts', async function (assert) {
+  test('it should hide secrets sync stats', async function (assert) {
     assert
-      .dom(CHARTS.chart('Secrets sync usage'))
-      .doesNotExist('chart is hidden because feature is not activated');
-
-    assert.dom('[data-test-stat-text="secret-syncs"]').doesNotExist();
+      .dom(CLIENT_COUNT.statTextValue('Secret sync'))
+      .doesNotExist('stat is hidden because feature is not activated');
+    assert.dom(CLIENT_COUNT.statTextValue('Entity')).exists('other stats are still visible');
```
Comment (noelledaley) [inline ui/tests/acceptance/clients/counts/overview-test.js:285]: we were previously making a `doesNotExist` assertion on the wrong element. oops! i fixed that throughout this file and added a related assertion to ensure we don't get a falsely passing tests (i.e. if _all_ the chart stats were missing for some reason).

File: ui/tests/integration/components/clients/page/sync-test.js [hunk 18]
```diff
@@ -26,22 +27,9 @@ module('Integration | Component | clients | Clients::Page::Sync', function (hook
   setupMirage(hooks);
 
   hooks.beforeEach(async function () {
-    clientsHandler(this.server);
```
Comment (noelledaley) [inline ui/tests/integration/components/clients/page/sync-test.js:29]: i promise i didn't actually delete this! i just needed to split the tests out into two modules because i needed to use the `syncHandler` for some tests and no handler for another. mirage doesn't behave well if you try to switch between handlers within the same module, nor does it work when you try to overwrite a handler already in use. therefore it was easiest to split these up.

File: ui/tests/acceptance/clients/counts/overview-test.js (hunk 15 above)
Comment (noelledaley) [reply ui/tests/acceptance/clients/counts/overview-test.js]: i looked at this for quite awhile today and these chart / chart filtering tests seem to be flaky; at least on our sidebranch. check out the video i left in slack for more (github wouldn't let me upload it here because the file size was too big 🙁 )

@Monkeychip and i talked about it while pairing today and assuming these tests are failing on other branches, we'll need to tackle the fixes in a separate PR.

File: ui/app/components/clients/page/counts.ts [hunk 19]
```diff
@@ -162,6 +164,22 @@ export default class ClientsCountsPageComponent extends Component<Args> {
     return activity?.total;
   }
 
+  get showSecretsSync(): boolean {
```
Comment (noelledaley) [inline ui/app/components/clients/page/counts.ts:167]: since this value is only needed on one component, we don't need to compute this in the route to add to the model. better off calculating it closer to where we actually use the value, where we already have enough context to compute it. in other words, by calculating the value here we can simply pass it straight into the `<NavBar />` which is the only place where it's used. 😀

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 20]
```diff
@@ -95,14 +68,11 @@ export default class ClientsCountsRoute extends Route {
...
     const { activity, activityError } = await this.getActivity(startTimestamp, endTimestamp);
 
-    const isSecretsSyncActivated = await this.isSecretsSyncActivated(activity);
-
     return {
       activity,
       activityError,
       config,
       endTimestamp,
-      isSecretsSyncActivated,
```
Comment (hellobontempo) [inline ui/app/routes/vault/cluster/clients/counts.ts:105]: Great idea moving this to the `counts.ts` component! Much clearer there than passing down from this route model hook 🥇

File: ui/tests/acceptance/clients/counts/overview-test.js (hunk 17 above)
Comment (hellobontempo) [reply ui/tests/acceptance/clients/counts/overview-test.js:285]: Great catch!

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 21]
```diff
@@ -314,9 +302,29 @@ module('Acceptance | clients | overview | sync not in license', function (hooks)
     assert.dom(GENERAL.tab('sync')).doesNotExist();
   });
 
-  test('it should hide secrets sync charts', async function (assert) {
-    assert.dom(CHARTS.chart('Secrets sync usage')).doesNotExist();
+  test('it should hide secrets sync stats', async function (assert) {
+    assert.dom(CLIENT_COUNT.statTextValue('Secret sync')).doesNotExist();
```
Comment (hellobontempo) [inline ui/tests/acceptance/clients/counts/overview-test.js:306]: In case it comes up in another review - this stat was intentionally labelled `Secret sync` (singular) because it's referring to the client type, not the secrets sync feature.

**Review by hellobontempo — APPROVED** [review]
Great work and fantastic test updates ⭐

File: ui/app/components/clients/counts/nav-bar.hbs
Comment (hellobontempo) [inline ui/app/components/clients/counts/nav-bar.hbs:1]: It looks like tests are failing because this file needs a copyright header 
```
{{!
  Copyright (c) HashiCorp, Inc.
  SPDX-License-Identifier: BUSL-1.1
~}}

File: ui/tests/integration/components/clients/counts/nav-bar-test.js
Comment (hellobontempo) [inline ui/tests/integration/components/clients/counts/nav-bar-test.js:1]: Even though it's a test file, this file also needs a copyright header
```
/**
 * Copyright (c) HashiCorp, Inc.
 * SPDX-License-Identifier: BUSL-1.1
 */
